vmdb/app/controllers/application_controller.rb

```diff
@@ -2168,6 +2168,10 @@ def prov_redirect(typ=nil)
         render_flash_not_applicable_to_model(typ)
         return
       end
+      if typ == "migrate" && !VmOrTemplate.supports_migrate?(vm_ids)
```

Comment:
We can probably add a class method `is_available?` that takes one or more ids.  Then we can move supports_migrate and others into the is_available structure.

If you want, we can pair on this.  I think it's important enough to get a good design going, and this is a great place to start since SCVMM is not a major entry point yet, and any mistakes we make won't kill every user.  This is extremely important for pluggability, especially with respect to the UI.  cc @blomquisg